src/pages/UserSite/HomePage/homePage.tsx

```diff
@@ -94,7 +95,7 @@ export default class HomePage extends React.Component{
         if(establishmentsSearch.length === app.state.establishments.length){
             return;
         }
-        establishmentsSearch.forEach((value:any, key:number)=>{
+        establishmentsSearch.forEach((value:any, key:number)=>{ // note
```

Comment:
Du bør ikke endre state på andre måter enn med `this.setState()` for at React skal registrere endringene og rendere på nytt. Det er heller ikke noen god ide å sette hele componenter i state, heller bruk dataen du har i state for å rendere componentene direkte. 

F.eks i `render()` der du har `<Row>{app.state.recommendedEstablishments}</Row>` nå, kan du bruke `state.recommendedEstablishments.map()`: 

```
<Row>
    {this.state.recommendedEstablishments.map(establishment => {
        <RecommendComp
             image={establishment.imageUrl}
             hotelName={establishment.establishmentName}
             id={establishment.id}
             key={establishment.id} // Ikke bruk array index som key, heller bruk IDen
         />
    })
</Row>
```

Jeg hadde satt `recommendedEstablishments` i state samtidig som du fetcher establishments i `componentDidMount()`